compliance/index/v4/convert.go

```diff
@@ -0,0 +1,112 @@
+package v4
+
+import (
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/clair"
+	"github.com/stackrox/rox/pkg/cvss/cvssv3"
+	"github.com/stackrox/rox/pkg/protocompat"
+)
+
+func ToNodeScan(r *v4.VulnerabilityReport) *storage.NodeScan {
+	return &storage.NodeScan{
+		ScanTime:       protocompat.TimestampNow(),
+		Components:     toStorageComponents(r),
+		Notes:          toStorageNotes(r.Notes),
+		ScannerVersion: storage.NodeScan_SCANNER_V4,
+	}
+}
+
+func toStorageComponents(r *v4.VulnerabilityReport) []*storage.EmbeddedNodeScanComponent {
+	result := make([]*storage.EmbeddedNodeScanComponent, 0)
+	vulnerabilities := r.GetVulnerabilities()
+	packages := r.GetContents().GetPackages()
+
+	for pkgID, pkgvuln := range r.GetPackageVulnerabilities() {
+		pkg := findPackage(packages, pkgID)
+		if pkg == nil {
+			log.Warnf("Unable to find package for id %s in %d packages. Skipping", pkg, len(packages))
+			continue
+		}
+		vulns := make([]*storage.EmbeddedVulnerability, 0)
+		for _, vulnID := range pkgvuln.GetValues() {
+			vuln := vulnerabilities[vulnID]
```

Comment:
I've looked at `VulnerabilityReport`, it has two fields `Vulnerabilities` and `PackageVulnerabilities`. My concern here is that vulnID ranged over the `PackageVulnerabilities` but accesses `Vulnerabilities` at those indices.

Are `Vulnerabilities` and `PackageVulnerabilities` always populated equally or is there the possibility that `vulnerabilities[vulnID]` could go out of range?